adapters/silvermob/silvermob.go

```diff
@@ -177,28 +177,36 @@ func (a *SilverMobAdapter) MakeBids(
 	bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
 
 	for _, sb := range bidResp.SeatBid {
-		for _, bid := range sb.Bid {
-			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
-				Bid:     &bid,
-				BidType: getMediaTypeForImp(bid.ImpID, openRTBRequest.Imp),
-			})
+		for i := range sb.Bid {
+			bid := sb.Bid[i]
+			bidType, err := getBidMediaTypeFromMtype(&bid)
+
+			if err != nil {
+				errs = append(errs, err)
+			} else {
+				bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+					Bid:     &bid,
+					BidType: bidType,
+				})
+			}
+
 		}
 	}
 
-	return bidResponse, nil
+	return bidResponse, errs
 }
 
-func getMediaTypeForImp(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
-	mediaType := openrtb_ext.BidTypeBanner
-	for _, imp := range imps {
-		if imp.ID == impId {
-			if imp.Video != nil {
-				mediaType = openrtb_ext.BidTypeVideo
-			} else if imp.Native != nil {
-				mediaType = openrtb_ext.BidTypeNative
-			}
-			return mediaType
-		}
+func getBidMediaTypeFromMtype(bid *openrtb2.Bid) (openrtb_ext.BidType, error) {
+	switch bid.MType {
+	case openrtb2.MarkupBanner:
+		return openrtb_ext.BidTypeBanner, nil
+	case openrtb2.MarkupVideo:
+		return openrtb_ext.BidTypeVideo, nil
+	case openrtb2.MarkupNative:
+		return openrtb_ext.BidTypeNative, nil
+	case openrtb2.MarkupAudio:
+		return openrtb_ext.BidTypeAudio, nil
+	default:
+		return "", fmt.Errorf("Unable to fetch mediaType for imp: %s", bid.ImpID)
```

Comment:
According to your YAML file you don't support audio which leads me to believe this case statement should be removed so that an audio mtype results in an error.

Reply:
Actually we do support audio, but I've removed for now this case statement. We will back it later when add tests and update yaml